index.js

```diff
@@ -332,21 +347,27 @@ class CoverageSonar extends CoverageBase {
      * @param {Object} config
      * @param {String} [config.scope]           Coverage scope
      * @param {Object} config.buildCredentials  Information stored in a build JWT
-     * @param {String} config.projectKey        Sonar project key
-     * @param {String} config.username          Sonar username
+     * @param {String} [config.jobName]         Screwdriver job name
+     * @param {String} config.pipelineName      Screwdriver pipeline name
+     * @param {String} [config.projectKey]      Sonar project key
+     * @param {String} [config.projectName]     Sonar project name
+     * @param {String} [config.username]        Sonar username
      * @return {Promise}                        An access token that build can use
      *                                          to talk to coverage server
      */
-    _getAccessToken({ scope, username, projectKey, projectName, buildCredentials }) {
+    _getAccessToken({ scope, username, projectKey, projectName, jobName, pipelineName,
+        buildCredentials }) {
         const { jobId, pipelineId, prParentJobId } = buildCredentials;
         let projectData = { username, projectKey, projectName };
 
-        if (!username || !projectKey || !projectName) {
+        if (!username || !projectKey || !projectName || projectName.includes('undefined')) {
```

Comment:
if projectName is a string should we be checking 
`typeof(projectName) == "undefined" `

Reply:
ah no was checking literal inclusion of string "undefined"